File: android/app/src/main/java/atomic1028/gl/emeraldknight/ui/AboutFragment.java
package atomic1028.gl.emeraldknight.ui;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import atomic1028.gl.emeraldknight.R;

public class AboutFragment extends Fragment {
    public AboutFragmentListener listener;

    public static interface AboutFragmentListener {
        void openCheat();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof AboutFragmentListener) {
            listener = (AboutFragmentListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.about_fragment, container, false);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        Button btn = getView().findViewById(R.id.cheat_button);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.openCheat();
            }
        });
    }
}

File: android/app/src/main/java/com/tuzicao/emeraldknight/ui/AboutFragment.kt
package com.tuzicao.emeraldknight.ui

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import com.tuzicao.emeraldknight.R

class AboutFragment : Fragment() {
    var listener: AboutFragmentListener? = null

    interface AboutFragmentListener {
        fun openCheat()
        fun openProject()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is AboutFragmentListener) {
            listener = context
        } else {
            throw RuntimeException("$context must implement AboutFragmentListener")
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.about_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val cheat_btn: Button = view.findViewById(R.id.cheat_button)
        cheat_btn.setOnClickListener {
            listener?.openCheat()
        }
        val page_btn:Button =view.findViewById(R.id.page_btn)
        page_btn.setOnClickListener{
            listener?.openProject()
        }
    }
}
